solution/backend/resources/views/resources.py

```diff
@@ -61,29 +62,39 @@ def get_annotated_url(self):
             default=None,
         )
 
-    def format_gov_results(self, gov_results):
-        if "errors" in gov_results or (gov_results["web"]["total"] == 0 and not gov_results["web"]["results"]):
-            raise NotFound("Invalid page")
+    def format_gov_results(self, gov_response):
         results = []
-        for gov_result in gov_results['web']['results']:
-            results.append({
-                'name': gov_result['title'],
-                'snippet': gov_result['snippet'],
-                'url': gov_result['url']
-            })
-        self.gov_results = {
-            "total": gov_results["web"]["total"],
-            "results": results,
-        }
+        if "errors" in gov_response or (gov_response["web"]["total"] == 0 and not gov_response["web"]["results"]):
+            self.gov_results = {
+                'total': 0,
+                'results': results
+            }
+        else:
+            for gov_result in gov_response['web']['results']:
+                results.append({
+                    'name': gov_result['title'],
+                    'snippet': gov_result['snippet'],
+                    'url': gov_result['url']
+                })
+            self.gov_results = {
+                "total": gov_response["web"]["total"],
+                "results": results,
+            }
 
     def get_gov_results(self, query, page):
         key = os.environ.get('SEARCHGOV_KEY')
         offset = (page - 1) * self.limit
 
         rstring = f'https://search.usa.gov/api/v2/search/?affiliate=reg-pilot-cms-test&access_key={key}' \
                   f'&query={urlparse.quote_plus(query)}&limit={self.limit}&offset={offset}'
-        gov_results = json.loads(requests.get(rstring).text)
-        self.format_gov_results(gov_results)
+        try:
+            gov_response = json.loads(requests.get(rstring, timeout=3).text)
+        except ValueError:
+            gov_response = "errors"
+        except ConnectTimeout:
+            gov_response = "errors"
```

Comment:
can we make the gov_response for ValueError different from ConnectionTimeout. This will allow us to troubleshoot in the logs in the future.

Reply:
They are different errors.  The timeout error happens when its over 3 seconds in the request.get.  The value error is if the page returns something funky that isnt json formattable.
json.loads(requests.get(rstring, timeout=3).text)

Reply:
This response is for the next part where we transform the data.  We are still getting an error regardless and it will be throwing the exception.  We are just handling it in order for the app to not stop.